portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryMetadataLocalServiceImpl.java

```diff
@@ -60,6 +60,18 @@ public void deleteFileEntryMetadata(long fileEntryId)
 		}
 	}
 
+	@Override
+	public void deleteFileEntryMetadataByDDMStructureId(long ddmStructureId)
+		throws PortalException {
+
+		for (DLFileEntryMetadata dlFileEntryMetadata :
+				dlFileEntryMetadataPersistence.findByDDMStructureId(
+					ddmStructureId)) {
+
+			deleteFileEntryMetadata(dlFileEntryMetadata);
+		}
```

Comment:
Why not use `removeByDDMStructureId` here?

Reply:
[deleteFileEntryMetadata](https://github.com/SamZiemer/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryMetadataLocalServiceImpl.java#L31-L47) also removes DDM storage and structure links associated with the fileEntryMetadataId, which I was not catching in the 		_ddmStructureLinkLocalService.deleteStructureStructureLinks(documentMetadataSetId) call.